src/com/owncloud/android/ui/activity/ManageSpaceActivity.java

```diff
@@ -0,0 +1,131 @@
+
+        ActionBar actionBar = getSupportActionBar();
+        actionBar.setDisplayHomeAsUpEnabled(true);
+        actionBar.setTitle(R.string.manage_space_title);
+
+        TextView descriptionTextView = (TextView) findViewById(R.id.general_description);
+        descriptionTextView.setText(getString(R.string.manage_space_description, getString(R.string.app_name)));
+
+        Button clearDataButton = (Button) findViewById(R.id.clearDataButton);
+        clearDataButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                clearData();
+            }
+        });
+    }
+
+    /**
+     * Save passcode from Share preferences
+     * Clear the rest of data
+     */
+    private void clearData() {
+        // Save passcode from Share preferences
+        SharedPreferences appPrefs = PreferenceManager
+                .getDefaultSharedPreferences(getApplicationContext());
+
+        boolean passCodeEnable = appPrefs.getBoolean("set_pincode", false);
+
+        String passCodeDigits[] = new String[4];
+        if (passCodeEnable) {
+            passCodeDigits[0] = appPrefs.getString("PrefPinCode1", null);
+            passCodeDigits[1] = appPrefs.getString("PrefPinCode2", null);
+            passCodeDigits[2] = appPrefs.getString("PrefPinCode3", null);
+            passCodeDigits[3] = appPrefs.getString("PrefPinCode4", null);
+        }
+
+        // Clear data
+        clearApplicationData();
+
+
+        // Clear SharedPreferences
+        SharedPreferences.Editor appPrefsEditor = PreferenceManager
+                .getDefaultSharedPreferences(getApplicationContext()).edit();
+        appPrefsEditor.clear();
+        appPrefsEditor.commit();
+
+        // Recover passcode
+        if (passCodeEnable) {
+            appPrefsEditor.putString("PrefPinCode1", passCodeDigits[0]);
+            appPrefsEditor.putString("PrefPinCode2", passCodeDigits[1]);
+            appPrefsEditor.putString("PrefPinCode3", passCodeDigits[2]);
+            appPrefsEditor.putString("PrefPinCode4", passCodeDigits[3]);
+        }
+
+        appPrefsEditor.putBoolean("set_pincode", passCodeEnable);
+        appPrefsEditor.commit();
+
+
+        finish();
+        System.exit(0);
+
+    }
+
+    public void clearApplicationData() {
+        File cache = getCacheDir();
+        File appDir = new File(cache.getParent());
+        if (appDir.exists()) {
+            String[] children = appDir.list();
+            for (String s : children) {
+                if (!s.equals("lib")) {
+                    File fileToDelete = new File(appDir, s);
+                    deleteDir(fileToDelete);
+                    Log_OC.d(TAG, "Clear Application Data, File: " + fileToDelete.getName()+ " DELETED *******");
+                }
+            }
+        }
+    }
+
+    public boolean deleteDir(File dir) {
+        if (dir != null && dir.isDirectory()) {
+            String[] children = dir.list();
```

Comment:
Thanks, @przybylski . Pretty interesting, indeed.

My source of worry: https://www.youtube.com/watch?v=MZOf3pOAM6A ; results around time 3:00, they refer to range based loop as "for simple". Maybe the penalty is not strictly for Java VMs, but a problem in Android VM.

Please, let me clarify.  I will not ask anybody to avoid range base loops in contributions, unless it results in a real performance problem in the app. It's just that I will not recommend either replacing index based loops with range based loops, nor consider index based loops as a reason to block or reject PRs.